assets/src/components/activities/common/authoring/AuthoringButton.tsx

```diff
@@ -0,0 +1,31 @@
+import { useAuthoringElementContext } from 'components/activities/AuthoringElement';
+import React, { MouseEventHandler } from 'react';
+import { classNames } from 'utils/classNames';
+
+export type Props = {
+  onClick: MouseEventHandler<HTMLButtonElement>;
+  style?: React.CSSProperties;
+  className?: string;
+  children?: React.ReactNode;
+  disabled?: boolean;
+  editMode: boolean;
+};
+
+export const AuthoringButton: React.FC<Props> = (props: Props) => {
+  return (
+    <button
+      style={props.style}
+      className={classNames(['btn', props.className])}
+      disabled={props.disabled || !props.editMode}
+      type="button"
+      onClick={props.onClick}
+    >
+      {props.children}
+    </button>
+  );
+};
+
+export const AuthoringButtonConnected: React.FC<Omit<Props, 'editMode'>> = (props) => {
+  const { editMode } = useAuthoringElementContext();
+  return <AuthoringButton {...props} editMode={editMode} />;
+};
```

Comment:
The new "connected" components can only be rendered within an `AuthoringElement` (which all activities are). They just pull the editmode/project slug from context so they don't have to be passed down through every component.